engine/src/main/java/io/zeebe/engine/processing/streamprocessor/ReProcessingStateMachine.java

```diff
@@ -330,8 +348,18 @@ private void updateStateUntilDone() {
   }
 
   private void onRecordReprocessed(final LoggedEvent currentEvent) {
-    if (currentEvent.getPosition() == lastSourceEventPosition) {
+    reprocessingStreamWriter.removeRecord(
+        currentEvent.getKey(), currentEvent.getSourceEventPosition());
+
+    // do reprocessing until the last source event but read until the last follow-up event to check
+    // for inconsistent reprocessing records
+    if (currentEvent.getPosition() >= lastFollowUpEventPosition) {
       LOG.info(LOG_STMT_REPROCESSING_FINISHED, currentEvent.getPosition());
+      LOG.debug("No reprocessing issue detected on partition {}.", zeebeState.getPartitionId());
```

Comment:
Is this log required? Previous log says it was successfully reprocessed.

Reply:
It is not required. I thought it is more explicit but I can also remove it.